[Doc] - Add entryFormat description and performance test class (#1694)
streamnative/kop: docs/configuration.md

### Choose the proper `entryFormat`

> The `entryFormat` values as below in the KoP.

Reviewer: ```suggestion
This table lists `entryFormat` values that are supported in KoP.
```

Author: Agree

| Name        | Description                                                                                                                                                                                                                                                                                           |
|-------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | pulsar      | The default `entryFormat` in the KoP. It's a tool to use for encoding/decoding between Kafka messages format and Pulsar messages format. So the performance is the worst. The advantage is both Kafka client and Pulsar client consumers can consume the messages from the Pulsar cluster.            |

Reviewer: ```suggestion
| pulsar      | `pulsar` is the default `entryFormat` in KoP. It is used to encode or decode formats between the Kafka message and the Pulsar message. Therefore, the performance is the worst. The benefit is that both the Kafka client and the Pulsar client can consume the messages from the Pulsar cluster.            |
```

Author: It's better than mine.

> | kafka       | If we set the `entryFormat=kafka`, the KoP will not encode/decode the Kafka messages. The messages will be stored in the bookie cluster in entries format. So the performance is the best. Because the messages keep the Kafka message format, the Pulsar client consumers can not parse the messages. |

Reviewer: ```suggestion
| kafka       | When you set the `entryFormat` option to `kafka`, KoP does not encode or decode Kafka messages. The messages will be directly stored in the bookie cluster in entries format, and the Pulsar client can not parse these messages. Therefore, the performance is the best. |
```

> | mixed_kafka | It likes the `kafka` entryFormat. But they have some differences. It supports some non-official Kafka client implementations. This format is compatible with other non-official clients when encoding/decoding Kafka messages. The performance is less than `kafka` format.                            |

Reviewer: ```suggestion
| mixed_kafka | The `mixed_kafka` format works similarly to the `kafka` format.  You can set this option for some non-official Kafka clients for encoding or decoding Kafka messages. The performance is medium.                            |
```
one more questions, "This format is compatible with other non-official clients" do you mean other Kafka client or Pulsar client or both of them?

Author: Please refer to 
```However, some non-official Kafka clients might not work for `kafka` format. For example, old [Golang Sarama](https://github.com/Shopify/sarama) client didn't assign relative offsets in compressed message sets before [Shopify/sarama #1002](https://github.com/Shopify/sarama/pull/1002). In this case, the broker has to assign relative offsets and then do recompression. Since this behavior leads to some performance loss, KoP adds the `mixed_kafka` format to perform the conversion. The `mixed_kafka` format should be chosen when you have such an old Kafka client. Like `kafka` format, in this case, Pulsar consumers still cannot consume messages from Kafka producers.```

Reviewer: Got it. Thanks.
